functions/src/CVE/index.ts

```diff
@@ -1,46 +1,89 @@
 import * as express from 'express';
 import * as bodyParser from "body-parser";
+import * as admin from 'firebase-admin';
+import * as Enumerable from 'linq';
 
 const app = express();
 const main = express();
 
 main.use(app);
 main.use(bodyParser.json());
 
-
 export const getCVE = functions.https.onRequest(main);
 
 app.get('/cves', (request, response) => {
+  
     const bulletinID = request.query.bulletinid;
-    if (bulletinID !== null){
-      //TODO: call helper function to query for bulletin data 
-    }
     const SPLID = request.query.splid;
-    if (SPLID !== null){
-      //TODO: call helper function to query for spl data
-    }
     const SPLStart = request.query.splstart;
-    if (SPLStart !== null){
-      //TODO: call helper function to query for spl start data 
+    const CVEID = request.query.cveid;
+    const SPL1 = request.query.spl1;
+    const SPL2 = request.query.spl2;  
+
+    if (bulletinID !== null && bulletinID !== undefined){
+      bulletinIDHelper(String(bulletinID),response);
     }
-    const CVEID = request.query.cveid; 
-    if (CVEID !== null){
-      //TODO: call helper function for cve id data 
+    else if (SPLID !== null && SPLID !== undefined){
+      SPLIDHelper(String(SPLID),response);
     }
-    const SPL1 = request.query.spl1;
-    const SPL2 = request.query.spl2;
-    if (SPL1 !== null && SPL2 !== null){
+    else if (SPLStart !== null && SPLStart !== undefined){
+      //TODO: call helper function to query for spl start data 
+    } 
+    else if (CVEID !== null && CVEID !== undefined){
+      CVEIDHelper(String(CVEID),response);
+    }
+    else if (SPL1 !== null && SPL2 !== null
+      && SPL1 !== undefined && SPL2 !== undefined){
       //TODO: call helper function for data in between spls
     }
 
-    response.send('Testing CVE get.');
-
 });
 
-//function bulletinIDHelper(id)
-//function SPLIDHelper(id)
+function SPLIDHelper(id:string,res:any){
+  const db = admin.database();
+  const ref = db.ref('/CVEs');
+  ref.once('value', function(snapshot) {
+    let cves = snapshot.val();
+    cves = Enumerable.from(cves)
+    .where(function (obj) { return obj.value.patch_level === id })
+    .select(function (obj) { return obj.value })
+    .toArray();
+    const result = {'CVEs': cves};
+    res.send(result);
+  }).catch(error => {
+    console.log("error getting CVEs for spl:"+ error);
+  });
+}
+
+function bulletinIDHelper(id:string,res:any){
+  const db = admin.database();
+  const ref = db.ref('/CVEs');
+  ref.once('value', function(snapshot) {
+    let cves = snapshot.val();
+    cves = Enumerable.from(cves)
+    .where(function (obj) { return obj.value.ASB === id })
+    .select(function (obj) { return obj.value })
+    .toArray();
+    const result = {'CVEs': cves};
+    res.send(result);
+  }).catch(error => {
+    console.log("error getting CVEs for bulletinID:"+ error);
+  });
+}
+
 //function SPLStartHelper(id)
-//function CVEIDHelper(id)
+
+function CVEIDHelper(id:any,res:any){
```

Comment:
try to avoid words like "helper", "handler", and "processor" in your function names, its generally not good style as its not descriptive of what the function is actually doing.

Reply:
> try to avoid words like "helper", "handler", and "processor" in your function names, its generally not good style as its not descriptive of what the function is actually doing.

I created a branch Apicalls-group2 off this branch and have been pushing other calls in it. Those other calls are also in spl/index.ts and cve/ts. I will dress these issues in the PR for that branch later today

Reply:
having another branch should not prevent addressing issues in a current pull request. The changes can be addressed in this pr, and the other branch can always be rebased off of it or master (once this pr is approved)

Reply:
> having another branch should never prevent addressing issues in a current pull request. The changes can be addressed in this pr, and the other branch can always be rebased off of it or master (once this pr is approved)

Understood!!